src/runtime-rs/crates/hypervisor/src/qemu/cmdline_generator.rs

```diff
@@ -2071,3 +2080,23 @@ fn get_devno_ccw(ccw_subchannel: &mut Option<CcwSubChannel>, device_name: &str)
         )
     })
 }
+
+#[derive(Debug)]
+struct DeviceVirtioBalloon {}
+
+impl DeviceVirtioBalloon {
+    fn new() -> Self {
+        DeviceVirtioBalloon {}
+    }
+}
+
+#[async_trait]
+impl ToQemuParams for DeviceVirtioBalloon {
+    async fn qemu_params(&self) -> Result<Vec<String>> {
+        Ok(vec![
+            "-device".to_owned(),
+            "virtio-balloon-pci,".to_owned(),
```

Comment:
All machine types don't necessarily rely on PCI to implement virtio transport, e.g. s390x uses CCW.
If you just pass `virtio-balloon`, QEMU will use the appropriate transport internally.

Reply:
Fixed.